nengo_spa/semantic_pointer.py

```diff
@@ -78,21 +93,25 @@ def _get_algebra(cls, vocab, algebra):
         return algebra
 
     def _get_unary_name(self, op):
-        return "{}({})".format(op, self.name) if self.name else None
+        return UnaryOperator(op, self._expr_tree) if self._expr_tree else None
 
     def _get_method_name(self, method):
-        return "({}).{}()".format(self.name, method) if self.name else None
+        return (
+            FunctionCall(tuple(), AttributeAccess(method, self._expr_tree))
+            if self._expr_tree
+            else None
+        )
 
     def _get_binary_name(self, other, op, swap=False):
         if isinstance(other, SemanticPointer):
-            other_name = other.name
+            other_expr_tree = other._expr_tree
         else:
-            other_name = str(other)
-        self_name = self.name
-        if self_name and other_name:
+            other_expr_tree = Leaf(str(other))  # FIXME what is this for?
```

Comment:
Do you want to resolve this before merging?

Reply:
oops 😶 Removed the comment, wrote it in a confused moment, but the `other` operand migth not be a `SemanticPointer` (could be a scalar for example), thus this is required.